web/model/floor.ts

```diff
@@ -1,24 +1,45 @@
-class Floor{
-    private length: number;
-    private width: number;
-    private SVG?: string;
-    constructor(length: number, width: number, svg?: string){
-        this.length = length;
-        this.width = width;
-        this.SVG = svg;
-    }
-
-    public getLength(): number {
-        return this.length;
-    }
-
-    public getWidth(): number {
-        return this.width;
-    }
-
-    public getSVG(): string | undefined{
-        return this.SVG;
-    }
+import { SVG } from "@/model/svg";
+
+class Floor {
+	private length: number;
+	private width: number;
+	private SVG: SVG;
+	constructor(length: number, width: number, svg: string) {
+		this.length = length;
+		this.width = width;
+		this.SVG = new SVG(svg, this.length, this.width);
+	}
+
+	clone() {
+		const floor = new Floor(
+			this.getLength(),
+			this.getWidth(),
+			this.getSVG().getString()
+		);
+		floor.getSVG().setLength(this.getSVG().getLength());
```

Comment:
qui anzichè fare così potresti mettere 

floor.setSVG(this.svg.clone())

implementando il metodo clone dentro SVG

Reply:
ci avevo pensato ad implementare un clone in SVG, ma:
invocando il costruttore di floor invoco anche il costruttore di svg, quindi già creo un'istanza di copia di svg (poi aggiusto i parametri con i setter perché di default ha gli stessi parametri di floor).
Il clone di svg dovrebbe ritornare un'istanza di svg da sostituire a quello già creato, che mi sembra un operazione inutile.
Avevo pensato anche alla possibilità di implementare un costruttore che accetta come parametri (number, number, SVG) ma a quanto pare non è una cosa comune in Typescript come in C++ (in più non sapevo se questa cosa potesse rompere o meno lo strategy).
Morale della favola ho deciso che visto che floor si occupa della creazione di svg allora non ci sono prolemi se si occupa anche della copia

Reply:
Ok però non capisco allora perchè non limitarsi a fare floor.setSVG(this.getFloor())

Come hai detto tu invochi il costruttore di floor e svg insieme, ma non vedo perchè tu non possa fare  floor.setSVG(this.getFloor()) invece di dover passare ogni volta per il set di una dimensione una alla volta.

Reply:
quindi dici spostare la modifica delle dimensioni del SVG da clone ad un metodo di copia di SVG che prende un SVG per parametro. Per me non c'è problema si può fare.